opm/material/common/UniformXTabulated2DFunction.hpp

```diff
@@ -42,6 +42,19 @@
 
 namespace Opm {
 
+
+
+/*!
+ * \brief Indicates how interpolation will be performed. Normal interpolation
+ *        is done by interpolating vertically between lines of sample points,
+ *        whereas LeftExtreme or RightExtreme implies guided interpolation, where
+ *        interpolation is done parallel to a guide line. With LeftExtreme
+ *        the lowest Y values will be used for the guide, and the guide line slope
+ *        extends unchanged to infinity. With RightExtreme, the highest Y values
+ *        are used, and the slope decreases linearly down to 0 at Y = 0.
+ */
+enum class InterpolationGuide { LeftExtreme, RightExtreme, Vertical };
```

Comment:
It was moved out of the class body to avoid a needless name nesting, but ok I'll move it inside and change the enum name. It definitely changes interpolation, not just extrapolation!

I do not think SlopeX is better, but I agree it is hard to find good names for the concepts.

Reply:
why does it affect *inter*polation? the code on line 312 clearly is for the case where the x value is larger than the biggest tabulated one?

Reply:
I do not see what you mean at all here. The x can be anywhere, you will get to that code as long as you are in RightExtreme mode.

Reply:
Okay, then I don't understand what this patch does. mind to explain it a bit more verbosely? (preferably by adding comments to the code where you calculate and use `shift`.)